pkgs/tools/system/netdata/default.nix

```diff
@@ -57,18 +60,20 @@ in stdenv.mkDerivation rec {
     ''}
   '';
 
-  preConfigure = optionalString (!stdenv.isDarwin) ''
+  preConfigure = optionalString withCloud ''
+    mkdir -p externaldeps
+    cp -r "${mosquitto}/lib" externaldeps/mosquitto;
+    cp -r "${libwebsockets_3_2}/lib" externaldeps/libwebsockets;
```

Comment:
Ugh https://github.com/netdata/netdata/blob/3d33eea007967ffe70f7918e6b617366cb34dd28/CMakeLists.txt#L916-L925

Does not someone (some other distro, pull request on the repo) have a patch that would allow using shared library from the system?

Reply:
I noticed that this is terrible practice by the upstream project, but I chose to simply adapt the build script for the initial PR.

Reply:
Checking out [other distros](https://repology.org/project/netdata/versions) (Alpine, Arch, Debian, Fedora, [Gentoo](https://gitweb.gentoo.org/repo/gentoo.git/tree/net-analyzer/netdata/netdata-1.23.2.ebuild?id=2eb0beba59cc05194c330ad0178a589f1605e902#n98)) I found a relevant issue:

https://github.com/netdata/netdata/issues/8961

Reply:
At least some of the developers think that accommodating Linux distributions is generally a good idea, however I am not that confident that we will see actual changes anytime soon. In any case, I have subscribed to the Issue and will of course update the PR if I notice any changes.

Reply:
[According to one of the developers](https://github.com/netdata/netdata/issues/8961#issuecomment-762337210), Netdata will soon be using a new implementation of the [Agent-Cloud-Link](https://github.com/netdata/netdata/pull/10315) which would only require unpatched and common dependencies (JSON-C and OpenSSL). I would be more than happy to modify this PR when this change lands, or if it could be agreed to merge this PR in its current "hackish" state, open a new PR to then fix these issues.

Reply:
@leotaku hi, you are right I am pushing for the new implementation of the link as you mentioned. Called `ACLK-NG`, started originally as a skunkworks project it is starting to take shape and we want to merge it ASAP.
While it works already it will take some time to weatherproof it and catch all possible bugs. This is the motive why originally it will be released as a fallback (in case ACLK-Legacy cannot be built e.g. due to missing custom mosquito lib), with the possibility to enforce its use by passing `--aclk-ng` to the installer.

The main motivation to do this for me was to enable full-featured Netdata in every distribution package manager (e.g. remove the need for bundling libraries and other "hacky" things). As a side effect, it also seems to perform better (throughput doesn't drop with latency as was the case with Legacy).